exporter/feature_event.go

```diff
@@ -33,38 +34,52 @@ func NewFeatureEvent(
 }
 
 // FeatureEvent represent an event that we store in the data storage
+// nolint:lll
 type FeatureEvent struct {
 	// Kind for a feature event is feature.
 	// A feature event will only be generated if the trackEvents attribute of the flag is set to true.
-	Kind string `json:"kind" example:"feature"`
+	Kind string `json:"kind" example:"feature" parquet:"name=kind, type=BYTE_ARRAY, convertedtype=UTF8"`
 
 	// ContextKind is the kind of context which generated an event. This will only be "anonymousUser" for events generated
 	// on behalf of an anonymous user or the reserved word "user" for events generated on behalf of a non-anonymous user
-	ContextKind string `json:"contextKind,omitempty" example:"user"`
+	ContextKind string `json:"contextKind,omitempty" example:"user" parquet:"name=contextKind, type=BYTE_ARRAY, convertedtype=UTF8"`
 
 	// UserKey The key of the user object used in a feature flag evaluation. Details for the user object used in a feature
 	// flag evaluation as reported by the "feature" event are transmitted periodically with a separate index event.
-	UserKey string `json:"userKey" example:"94a25909-20d8-40cc-8500-fee99b569345"`
+	UserKey string `json:"userKey" example:"94a25909-20d8-40cc-8500-fee99b569345" parquet:"name=userKey, type=BYTE_ARRAY, convertedtype=UTF8"`
 
 	// CreationDate When the feature flag was requested at Unix epoch time in milliseconds.
-	CreationDate int64 `json:"creationDate" example:"1680246000011"`
+	CreationDate int64 `json:"creationDate" example:"1680246000011" parquet:"name=creationDate, type=INT64"`
 
 	// Key of the feature flag requested.
-	Key string `json:"key" example:"my-feature-flag"`
+	Key string `json:"key" example:"my-feature-flag" parquet:"name=key, type=BYTE_ARRAY, convertedtype=UTF8"`
 
 	// Variation  of the flag requested. Flag variation values can be "True", "False", "Default" or "SdkDefault"
 	// depending on which value was taken during flag evaluation. "SdkDefault" is used when an error is detected and the
 	// default value passed during the call to your variation is used.
-	Variation string `json:"variation" example:"admin-variation"`
+	Variation string `json:"variation" example:"admin-variation" parquet:"name=variation, type=BYTE_ARRAY, convertedtype=UTF8"`
 
 	// Value of the feature flag returned by feature flag evaluation.
-	Value interface{} `json:"value"`
+	Value interface{} `json:"value" parquet:"name=value, type=BYTE_ARRAY, convertedtype=UTF8"`
```

Comment:
If I understand correctly, everything will be a `string` in the parquet file?

Reply:
yes the value will always be string
parquet's user may need to unmarshal it first to process the value